File: src/Servers/PresenceSearchPlayer/test/MokeObject.cs
using System.Net;
using Moq;
using UniSpy.Server.PresenceSearchPlayer.Application;
using UniSpy.Server.Core.Abstraction.Interface;

namespace UniSpy.Server.PresenceSearchPlayer.Test
{
    public class MokeObject
    {
        public static IClient Client = CreateClient();

        public static Client CreateClient(string ipAddress = "192.168.1.1", int port = 9999)
        {
            var managerMock = new Mock<IConnectionManager>();
            var connectionMock = new Mock<ITcpConnection>();
            connectionMock.Setup(s => s.RemoteIPEndPoint).Returns(new IPEndPoint(IPAddress.Parse(ipAddress), port));
            connectionMock.Setup(s => s.Manager).Returns(managerMock.Object);
            connectionMock.Setup(s => s.ConnectionType).Returns(NetworkConnectionType.Tcp);
            var serverMock = new PresenceSearchPlayer.Application.Server(managerMock.Object);

            return new Client(connectionMock.Object, serverMock);
        }
    }
}

File: src/servers/presence_connection_manager/tests/moke_object.py
import unittest
from unittest.mock import Mock
from servers.presence_connection_manager.applications.client import Client

class MokeObject:
    client = None

    # TODO We first need Server implementation
    @staticmethod
    def create_client(ip_address="192.168.1.1", port=9990):
        manager_mock = Mock(IConnectionManager)
        connection_mock = Mock()
        connection_mock.RemoteIPEndPoint = (ip_address, port)
        connection_mock.Manager = manager_mock
        connection_mock.ConnectionType = "Tcp"
        server_mock = Server(manager_mock)

        return Client(connection_mock, server_mock)
